src/plugins/unified_search/public/query_string_input/query_bar_top_row.tsx

```diff
@@ -789,7 +789,7 @@ export const QueryBarTopRow = React.memo(
               )}
               <EuiFlexItem
                 grow={!shouldShowDatePickerAsBadge()}
-                style={{ minWidth: shouldShowDatePickerAsBadge() ? 'auto' : 320, maxWidth: '100%' }}
+                css={{ minWidth: shouldShowDatePickerAsBadge() ? 'auto' : 320, maxWidth: '100%' }}
```

Comment:
should we create instead 2 css and switch them there? I'm not sure if this creates one everytime a new class whenever this `shouldShowDatePickerAsBadge` change.

something like
```
const showDatePickerAsBadge = css({minWidth: 'auto', maxWidth: '100%'};
const dontShowDatePickerAsBadge = css({minWidth: 320, maxWidth: '100%'};
...
css={shouldShowDatePickerAsBadge() showDatePickerAsBadge : dontShowDatePickerAsBadge}}
```

Reply:
> I'm not sure if this creates one everytime a new class

Emotion is smart enough not to do this generally, it creates the hash based on the CSS content, so it'll toggle between the two `className`s here. In general I do prefer statically defining CSS styles for performance, but this shouldn't be a huge issue here.